Env12

ggaffield opened the pull request:
This PR adds env12 to the search (refseq98+env12) and genome context displays.
The search was a simple change. I just pointed DATABASE to a new IGR file that has env12 concatenated to refseq98.
The genome context change is more involved, but doesn't change much of the original flow of genome_context.py.
I created an API on BL that returns a gzip'd GFF file when passed an accession number.  This hooks in rather simply to
the rest of the logic.  Though I did have to shuffle things around a little more than I expected to make it work smoothly.
"start.sh" now prompts for a BL API key, but indicates it's only required for Breaker Lab members.  If the BL key variable is not set,
DIMPL works just as is had before, and won't through any errors.
The key can be retrieved using this URL: http://bl.biology.yale.edu/lab/scripts/dimpl_getkey.pl

I only changed these 3 files:
    start.sh
    src/data/genome_context.py
    src/shell/infernal_source_template.sh

I made sure only those 3 files were staged in my commit.
I don't know why it's show 9 files.  Maybe the merge of master into my branch did it.
Can you just ignore (not merge) the others while processing the pull request?
If not, let me know and I'll see if I can figure out how to do it right.

kenibrewer commented:
@ggaffield I took a closer look at these changes, and I think the way you set this up is actually fine to include in the general purpose release of DIMPL even though the functionality will only work for Breaker Lab members. Your original pull request was a bit messy and included lots of inadvertent changes to the notebooks. Now that I've merged some of your other pull requests, these inadvertent changes are creating conflicts that prevent a clean merge. Would you mind cleaning this up and resubmitting?

Commit:
Added the "env12" environmental dataset to the search and genome context display.
** Only Breaker Lab members have access. Yale VPN & API key required **
Also: Improved the robustness of the Entrez transfers in genome_context.py.
